core/domain/collection_services.py

```diff
@@ -62,6 +63,28 @@
 _DEFAULT_RANK = 20
 
 
+def collection_node_metadata_dicts(query_string, search_cursor):
```

Comment:
Also I'm not sure why we need to touch collection_services at all. Can all this logic move directly into summary_services?